### PR TITLE
SAM aka Segment Anything Model

### PR DESCRIPTION
**Pull request by JulietteMoreau**
Review for the 13th april on SAM, the new META model released last Thursday.

### CODE REVIEW

**Comment by nathanpainchaud** [comment]
@JulietteMoreau Great choice of paper to review! I heard about this paper earlier today and I was going to check it out on my own, but now I'll just wait for your review :wink:

File: collections/_posts/2023-04-11-SAM.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+The goal of SAM is to be used in composition system (like CLIP in DALL-E) in order to get new capabilities that were not even thought about at its creation. To do so, SAM needs to predict a valid mask for a wide range of segmentation
+prompts in order to create a good interface between SAM and other components.
+
+It still has some limitations, there are some errors missing small structures or creating small disconnected components, and does not produce boundaries as crisply as more computationally intensive methods. Given the fact that it is a very generic segmentation model which goal is to perform with zero transfer learning, it is expected that all dedicated interactive segmentation methods to outperform SAM. Another limitation is the computational time: it is said to be real-time but it is not really the case when using a heavy image encoder. Finally even if SAM performs many tasks, it is not easy to design simple prompts that implements semantic and panoptic segmentation.
```
Comment (Volkaa) [inline collections/_posts/2023-04-11-SAM.md:152]:
```suggestion
It still has some limitations, there are some errors missing small structures or creating small disconnected components and does not produce boundaries as crisply as more computationally intensive methods. Given the fact that it is a very generic segmentation model whose goal is to perform with zero transfer learning, it is expected that all dedicated interactive segmentation methods outperform SAM. Another limitation is the computational time: it is said to be real-time but it is not really the case when using a heavy image encoder. Finally even if SAM performs many tasks, it is not easy to design simple prompts that implement semantic and panoptic segmentation.
```

**Review by Volkaa — CHANGES_REQUESTED** [review]
Very nice review, a very interesting recent article. Let's hope we won't be au chomage by the end of the PhD.

**Review by Volkaa — APPROVED** [review]

File: collections/_posts/2023-04-11-SAM.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+The ablations concern the SA-1B dataset and the architecture of SAM.
+
+![](/collections/images/SAM/ablation_study.jpg)
+
+The performance increase through all the data engine steps but in the end only images with automatically generated masks are kept to simplify setup training.
+
+# Discussion
+
+Models are almost always dedicated to a particular task, but here the foundation model trained on a very large dataset is then adapted to many tasks. However, generally foundation models are auto-supervised wherease here it is initialised with MAE that is based on a supervised work.
```
Comment (nathanpainchaud) [inline collections/_posts/2023-04-11-SAM.md:147]: wherease -> whereas
+
To clear the confusion with MAE being supervised or not as pointed out by Gaspard, I would suggest to rephrase things in the folloing way:
_it is initialised with a MAE trained using supervised labels._

**Review by nathanpainchaud — COMMENTED** [review]
A reminder of some typos/confusing phrasing pointed out during the presentation.

**Pull request merged**